lib/livereload.js

```diff
@@ -6,126 +6,116 @@
+/** Start the BrowserSync server and start livereloading
+ *  @param {string} projectRoot - The root of the cordova project being dealt with.
+ *  @param {Array.<string[]>} platforms - The platforms we want to livereload.
+ *  @param options - Options
+ *  @param options.ignore - files to ignore.
+ *  @param options.tunnel - use the --tunnel option to get around firewall/proxy issues.
+ *  @param options.ghostMode - sync gestures (form inputs, clicks, scrolls) across multiple devices.
+ *  @param options.cb - callback function to be called when .
  */
-function LiveReload(projectRoot, platformWWWs, options) {
-    this.projectRoot = projectRoot;
-    this.startPage = helpers.GetStartPage(this.projectRoot);
-    this.www_dir = path.join(this.projectRoot, 'www');
-    platformWWWDirs = platformWWWs;
-    this.startServer = startServer;
-    this.stopServer = stopServer;
-    this.options = options;
-    this.browserSync = require('browser-sync').create(); // If this is exposed to users, how would they handle deps install ?
-    return this;
-};
-
-
-function startServer() {
-    var self = this;
-    var deferred = Q.defer();
+module.exports.start = function (projectRoot, platforms, options) {
+    
+    // Validate whether all platforms are currently supported
+    // If a platform is not supported by the plugin, display an error message and don't process it any further
+    platforms.forEach(function (plat, index, platforms) {
+        if (!multiPlatforms.isPlatformSupported(plat)) {
+            var msg = 'The "' + plat + '" platform is not supported.';
+            livereloadEvents.emit('livereload:warning', msg);
+            platforms.splice(index, 1);
+        }
+    });
 
-    var defaultOptions = {
-        server: {
-            baseDir: self.projectRoot,
-            directory: true
-        },
-        open: false,
+    // If none of the supplied platforms are supported, stop and return an error
+    if (platforms === undefined || platforms.length === 0) {
+        var msg = 'None of the platforms supplied are currently supported for LiveReload.';
+        livereloadEvents.emit('livereload:error', msg);
+        return Q.reject(msg);
+    }
+
+    var BrowserSyncServer = require('./browserSyncServer');
+        
+    // If user has entered whitespaces surrounding '--ignore', 
+    // ... we end up having helpers.parseOptions() return 'ignore' with value true instead of a string,
+    // ... which can't be used as a string. So, Let's check for those cases
+    // e.g of error cases: 
+    //      `cordova run android -- --livereload --ignore, // with no path to ignore
+    //      `cordova run android -- --livereload --ignore= css, // with space after the '=' sign
+    //      etc...
+    // ToDO: ignore whitespaces surrounding the --ignore option or at least display a warning message to user
+    var ignoreOption = undefined;
+    if (options.ignore) {
+        ignoreOption = path.join(projectRoot, 'www', options.ignore);
```

Comment:
It would be nice to be able to ignore more than one directory.
I've created an issue to track that:  https://github.com/omefire/cordova-plugin-livereload/issues/45